core/src/main/java/com/opensymphony/xwork2/config/providers/XmlConfigurationProvider.java

```diff
@@ -777,11 +777,21 @@ protected Class verifyResultType(String className, Location loc) {
                 }
                 params.putAll(resultParams);
 
-                ResultConfig resultConfig = new ResultConfig.Builder(resultName, resultClass)
-                        .addParams(params)
-                        .location(DomHelper.getLocationObject(element))
-                        .build();
-                results.put(resultConfig.getName(), resultConfig);
+                Set<String> resultNamesSet;
+                if (",".equals(resultName.trim())) {
+                    resultNamesSet = new HashSet<>(1);
+                    resultNamesSet.add(resultName);
```

Comment:
You mean like this?

    Set<String> resultNamesSet = TextParseUtil.commaDelimitedStringToSet(resultName);
    if (resultNamesSet.isEmpty()) {
        resultNamesSet.add(resultName);
    }

Yeah, looks a bit cleaner to me. I'll change it.